integration-tests/src/main/java/io/druid/testing/ConfigFileConfigProvider.java

```diff
@@ -53,69 +55,94 @@ private void loadProperties(String configFile)
+        {
+        }
       );
     }
     catch (IOException ex) {
       throw new RuntimeException(ex);
     }
     // there might not be a router; we want routerHost to be null in that case
-    routerHost = props.get("router_host");
-    if (null != routerHost) {
-	routerHost += ":" + props.get("router_port");
+    routerUrl = props.get("router_url");
+    if (routerUrl == null) {
+      String routerHost = props.get("router_host");
+      if (null != routerHost) {
+        routerUrl = String.format("http://%s:%s", routerHost, props.get("router_port"));
+      }
+    }
+    brokerUrl = props.get("broker_url");
+    if (brokerUrl == null) {
+      brokerUrl = String.format("http://%s:%s", props.get("broker_host"), props.get("broker_port"));
+    }
+
+    historicalUrl = props.get("historical_url");
+    if (historicalUrl == null) {
+      historicalUrl = String.format("http://%s:%s", props.get("historical_host"), props.get("historical_port"));
+    }
+
+    coordinatorUrl = props.get("coordinator_url");
+    if (coordinatorUrl == null) {
+      coordinatorUrl = String.format("http://%s:%s", props.get("coordinator_host"), props.get("coordinator_port"));
+    }
+
+    indexerUrl = props.get("indexer_url");
+    if (indexerUrl == null) {
+      indexerUrl = String.format("http://%s:%s", props.get("indexer_host"), props.get("indexer_port"));
     }
-    brokerHost = props.get("broker_host") + ":" + props.get("broker_port");
-    historicalHost = props.get("historical_host") + ":" + props.get("historical_port");
-    coordinatorHost = props.get("coordinator_host") + ":" + props.get("coordinator_port");
-    indexerHost = props.get("indexer_host") + ":" + props.get("indexer_port");
     middleManagerHost = props.get("middlemanager_host");
+
     zookeeperHosts = props.get("zookeeper_hosts");
-    kafkaHost = props.get("kafka_host") + ":" + props.get ("kafka_port");
-
-    LOG.info ("router: [%s]", routerHost);
-    LOG.info ("broker: [%s]", brokerHost);
-    LOG.info ("coordinator: [%s]", coordinatorHost);
-    LOG.info ("overlord: [%s]", indexerHost);
-    LOG.info ("middle manager: [%s]", middleManagerHost);
-    LOG.info ("zookeepers: [%s]", zookeeperHosts);
-    LOG.info ("kafka: [%s]", kafkaHost);
+    kafkaHost = props.get("kafka_host") + ":" + props.get("kafka_port");
+
+    username = props.get("username");
+
+    password = props.get("password");
+
+    LOG.info("router: [%s]", routerUrl);
+    LOG.info("broker: [%s]", brokerUrl);
+    LOG.info("coordinator: [%s]", coordinatorUrl);
+    LOG.info("overlord: [%s]", indexerUrl);
+    LOG.info("middle manager: [%s]", middleManagerHost);
+    LOG.info("zookeepers: [%s]", zookeeperHosts);
+    LOG.info("kafka: [%s]", kafkaHost);
+    LOG.info("Username: [%s]", username);
   }
 
   @Override
   public IntegrationTestingConfig get()
   {
     return new IntegrationTestingConfig()
     {
+
       @Override
-      public String getCoordinatorHost()
+      public String getCoordinatorUrl()
```

Comment:
we have additional tests written internally which utilize these methods and things are failing to compile with this patch.